src/main/java/com/bento909/FileBasedLeagueGenerator.java

```diff
@@ -12,12 +12,48 @@
 
 public class FileBasedLeagueGenerator implements LeagueGenerator {
 
+    private static final int HOME_TEAM_NAME_INDEX = 0;
+    private static final int OPPOSITION_TEAM_NAME_INDEX = 1;
+    private static final int GOALS_FOR_INDEX = 2;
+    private static final int GOALS_AGAINST_INDEX = 3;
+    private static final int VENUE_INDEX = 4;
+    private static final int DATE_MATCH_PLAYED_INDEX = 5;
+    private static final int MAN_OF_THE_MATCH_INDEX = 6;
+    private CsvParser csvParser;
+    private SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd/MM/yy");
 
-    public FileBasedLeagueGenerator(String filename){
+    public FileBasedLeagueGenerator(String filename) {
+        csvParser = new CsvParser(filename);
     }
 
     @Override
     public League generateLeague() throws IOException, ParseException {
-        return null;
+        List<List<String>> fileContents = csvParser.parseFile();
+        Map<String, List<Match>> teamMap = new HashMap<>();
```

Comment:
The ```teamMap``` and the ```for``` loop below make sense as their own method. You should extract the logic to a private method.